pkg/agent/dispatcher.go

```diff
@@ -64,9 +64,7 @@ type Dispatcher struct {
 
 // Create a new Dispatcher, establishing a connection with the informant.
 // Note that this does not immediately start the Dispatcher. Call Run() to start it.
-func NewDispatcher(logger *zap.Logger, addr string, parent *InformantServer) (disp *Dispatcher, _ error) {
-	ctx := context.TODO()
-
+func NewDispatcher(ctx context.Context, logger *zap.Logger, addr string, parent *InformantServer) (disp *Dispatcher, _ error) {
```

Comment:
I noticed that the context this is called with is a long-lived context for the `serveInformantLoop` background worker.

As a follow-up PR, maybe we should create an inner context with a timeout in case connecting stalls?

Reply:
Sounds like a good idea, I'll add it to my list of todos.